GetConversion yields incorrect result on collection expression with spread in error case

See `MissingCtor_GenericIEnumerable_Spread` in comparison to `MissingCtor_GenericIEnumerable`.
This issue is referenced in tests

```

        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69521")]
        public void MissingCtor_GenericIEnumerable()
        {
            string source = """
                using System.Collections;
                using System.Collections.Generic;

                C x = [1]; // 1
                C.M([1]); // 2

                class C : IEnumerable<int>
                {
                    public C(string s) { }
                    IEnumerator IEnumerable.GetEnumerator() => null;
                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
                    public void Add(int i) { }

                    public static void M(C c) { }
                    public static void M(int[] i) { }
                }
                """;

            var comp = CreateCompilation(source).VerifyEmitDiagnostics(
                // (4,7): error CS7036: There is no argument given that corresponds to the required parameter 's' of 'C.C(string)'
                // C x = [1]; // 1
                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "[1]").WithArguments("s", "C.C(string)").WithLocation(4, 7),
                // (5,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(C)' and 'C.M(int[])'
                // C.M([1]); // 2
                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(C)", "C.M(int[])").WithLocation(5, 3)
                );

            var tree = comp.SyntaxTrees.First();
            var model = comp.GetSemanticModel(tree);
            var collections = tree.GetRoot().DescendantNodes().OfType<CollectionExpressionSyntax>().ToArray();

            var conversion1 = model.GetConversion(collections[0]);
            Assert.True(conversion1.IsValid);
            Assert.True(conversion1.IsCollectionExpression);

            var conversion2 = model.GetConversion(collections[1]);
            Assert.True(conversion2.IsValid);
            Assert.True(conversion2.IsIdentity);
            Assert.Null(model.GetTypeInfo(collections[1]).Type);
        }

        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69521")]
        public void MissingCtor_GenericIEnumerable_Spread()
        {
            string source = """
                using System.Collections;
                using System.Collections.Generic;

                int[] values = [1];
                C x = [..values]; // 1
                C.M([..values]); // 2

                class C : IEnumerable<int>
                {
                    public C(string s) { }
                    IEnumerator IEnumerable.GetEnumerator() => null;
                    IEnumerator<int> IEnumerable<int>.GetEnumerator() => null;
                    public void Add(int i) { }

                    public static void M(C c) { }
                    public static void M(int[] i) { }
                }
                """;

            var comp = CreateCompilation(source).VerifyEmitDiagnostics(
                // (5,7): error CS7036: There is no argument given that corresponds to the required parameter 's' of 'C.C(string)'
                // C x = [..values]; // 1
                Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "[..values]").WithArguments("s", "C.C(string)").WithLocation(5, 7),
                // (6,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(C)' and 'C.M(int[])'
                // C.M([..values]); // 2
                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(C)", "C.M(int[])").WithLocation(6, 3)
                );

            var tree = comp.SyntaxTrees.First();
            var model = comp.GetSemanticModel(tree);
            var collections = tree.GetRoot().DescendantNodes().OfType<CollectionExpressionSyntax>().ToArray();

            var conversion1 = model.GetConversion(collections[0]);
            Assert.True(conversion1.IsValid);
            Assert.True(conversion1.IsCollectionExpression);

            var conversion2 = model.GetConversion(collections[1]);
            Assert.True(conversion2.IsValid);
            Assert.True(conversion2.IsCollectionExpression);
            Assert.Null(model.GetTypeInfo(collections[1]).Type);
        }

```

--- a/src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs
+++ b/src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs
@@ -23357,8 +23357,8 @@ partial class Program
             Assert.True(conversion1.IsValid);
             Assert.True(conversion1.IsCollectionExpression);
 
-            // Note: we should probably not be getting a valid & collection conversion here
-            // Tracked by https://github.com/dotnet/roslyn/issues/70217
+            // Valid identity conversion matches the behavior for an ambiguous call
+            // with a switch expression argument: C.M(2 switch { _ => default });
             var conversion2 = model.GetConversion(collections[1]);
             Assert.True(conversion2.IsValid);
             Assert.True(conversion2.IsIdentity);
@@ -23372,7 +23372,7 @@ partial class Program
                 using System.Collections;
                 using System.Collections.Generic;
 
-                int[] values = [1];
+                int[] values = new int[] { 1 };
                 C x = [..values]; // 1
                 C.M([..values]); // 2
 
@@ -23405,11 +23405,11 @@ partial class Program
             Assert.True(conversion1.IsValid);
             Assert.True(conversion1.IsCollectionExpression);
 
-            // Note: we should not be getting a collection conversion here (see test above for contrast)
-            // Tracked by https://github.com/dotnet/roslyn/issues/70217
+            // Valid identity conversion matches the behavior for an ambiguous call
+            // with a switch expression argument: C.M(2 switch { _ => default });
             var conversion2 = model.GetConversion(collections[1]);
             Assert.True(conversion2.IsValid);
-            Assert.True(conversion2.IsCollectionExpression);
+            Assert.True(conversion2.IsIdentity);
             Assert.Null(model.GetTypeInfo(collections[1]).Type);
         }
 

